### ISSUE TITLE
OPD2 Local Repository management

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#382]
**Reported by wesbland on 3 Aug 2007 17:14 UTC**
OPD2 does not autodetect local repositories without an XML file.  This would be a very handy feature.

**Comment by dikim33** [comment]
**Comment by valleegr on 9 Dec 2007 21:35 UTC**
This should be fixed with the current version of trunk. The current version of OPD2 now uses PackagePath which provides a generic way to deal with repositories. It does not use any more the old concept of repository description via an XML file.

More testing is needed before to close the bug, but the XML file should not be used anymore.

**Comment by dikim33** [comment]
**Comment by valleegr on 10 Jan 2008 15:26 UTC**
The current implementation of OPD2 generates an error when trying to deal with local repositories (tested in interactive mode):

```
opd> l

$VAR1 = 'centos-5-x86_64';
$VAR2 = {
          'url' => '/tftpboot/distro/centos-5-x86_64.url',
          'oscar_repo' => 'file:/tftpboot/oscar/rhel-5-x86_64,file:/tftpboot/oscar/common-rpms',
          'distro_repo' => 'file:/tftpboot/distro/centos-5-x86_64',
          'os' => {
                    'compat_distro' => 'rhel',
                    'compat_distrover' => '5',
                    'distro' => 'centos',
                    'pkg' => 'rpm',
                    'arch' => 'x86_64',
                    'distro_version' => '5'
                  }
        };
$VAR3 = 'debian-4-x86_64';
$VAR4 = {
          'url' => '/tftpboot/distro/debian-4-x86_64.url',
          'oscar_repo' => 'file:/tftpboot/oscar/debian-4-x86_64,file:/tftpboot/oscar/common-debs,http://oscar.gforge.inria.fr/debian/+stable+main+oscar',
          'distro_repo' => 'http://bear.csm.ornl.gov/debian/+etch+main',
          'os' => {
                    'compat_distro' => 'debian',
                    'compat_distrover' => '4',
                    'distro' => 'debian',
                    'pkg' => 'deb',
                    'arch' => 'x86_64',
                    'distro_version' => '4'
                  }
        };
[ ]
Array: 0 element(s)
```

Clearly the result should not be an empty array.

**Comment by dikim33** [comment]
**Comment by valleegr on 14 Jan 2009 04:42 UTC**
OPD2 is not used anymore, instead we use ORM. But the problem remain the same.

**Comment by dikim33** [comment]
**Comment by valleegr on 15 Apr 2009 02:04 UTC**
(In [This check-in fixes #252, #512, #382, #513 - Deeply revisit ORM:
- Revisit the non-interactive mode to center the capabilities around the
notion of "distribution" (which is actually the level of abstraction ORM is
supposed to provide).
- [Ticket #252](8398])) Local caches are not anymore used, we directly query
repositories (modification due to the fact that we now deal with both online
and local repositories).
- [#512](Ticket) ORM is now fully based on the abstraction of Linux
distribution and we can now select different Linux distributions and query
them.
- [#382](Ticket) It is now possible to query repositories in interactive mode
(based on the current distribution).
- [#513](Ticket) Add an option to the interactive mode ("info") to display
repositories associated to the current Linux distribution.
